CRM/Contribute/Tokens.php

```diff
@@ -194,7 +194,43 @@ public function evaluateToken(TokenRow $row, $entity, $field, $prefetch = NULL)
       $row->tokens($entity, $field, $fieldValue);
     }
     else {
-      $row->dbToken($entity, $field, 'CRM_Contribute_BAO_Contribution', $field, $fieldValue);
+      // This section is copied & pasted & can be cleaned up later.
+      // most if probably already done in this functions
+      $tokenEntity = $entity;
+      $tokenField = $field;
+      $baoName = $this->getBAOName();
+      $baoField = $field;
+      if ($fieldValue === NULL || $fieldValue === '') {
+        $row->tokens($tokenEntity, $tokenField, '');
+      }
+
+      $fields = $baoName::fields();
+      if (!empty($fields[$baoField]['pseudoconstant'])) {
```

Comment:
this code will be removed in favour of the code a few lines above this chunk once the parameters are fully reconciled